backend/src/main/java/ai/verta/modeldb/experiment/FutureExperimentDAO.java

```diff
@@ -381,4 +391,96 @@ public FutureExperimentDAO(
               executor);
     }
   }
+
+  public InternalFuture<Experiment> updateExperimentNameOrDescription(
+      UpdateExperimentNameOrDescription request) {
+    return getProjectIdByExperimentId(Collections.singletonList(request.getId()))
+        .thenCompose(
+            projectIdFromExperimentMap ->
+                futureProjectDAO.checkProjectPermission(
+                    projectIdFromExperimentMap.get(request.getId()), ModelDBServiceActions.UPDATE),
+            executor)
+        .thenCompose(
+            unused -> {
+              var name = request.getName();
+              if (request.getName().isEmpty()) {
+                name = MetadataServiceImpl.createRandomName();
+              }
+
+              if (!name.isEmpty()) {
+                return updateExperimentName(
+                    request.getId(), ModelDBUtils.checkEntityNameLength(name));
+              }
+              // FIXME: this code never allows us to set the description as an empty string
+              if (!request.getDescription().isEmpty()) {
+                return updateExperimentDescription(request.getId(), request.getDescription());
+              }
+              return InternalFuture.completedInternalFuture(null);
+            },
+            executor)
+        .thenCompose(unused -> getExperimentById(request.getId()), executor);
+  }
+
+  private InternalFuture<Void> updateExperimentName(String expId, String name) {
+    return jdbi.useHandle(
+        handle ->
+            handle
+                .createUpdate(
+                    "update experiment set name = :name, date_updated = :updatedTime, version_number=(version_number + 1) where id = :id ")
```

Comment:
Similar to @conradoverta 's question one of the other PRs.  Does `name` need to be unique for all non-deleted experiments?

Reply:
Added unique experiment name check in entire project.